### PR TITLE
feat(web): in-app API documentation — REST + CLI + MCP coverage (replaces #565)

### PR DESCRIPTION
**Pull request by ZdenekSrotyr**
**Replaces [#565](https://github.com/keboola/agnes-the-ai-analyst/pull/565)** — extends Petr's PR ([@pcernik-grpn](https://github.com/pcernik-grpn)) to satisfy the project's triple-surface coverage policy and to close Devin Review's BUG_0001 from his PR.

## What

Carries every commit from #565 verbatim plus one additional release-cut commit on top that adds the two missing surfaces and the bug fix. Petr's curated guide, in-app `/documentation/api` page, Documentation nav section, and CI coverage gate are unchanged.

## Why a replacement PR

#565 is from a fork (`pcernik-grpn/agnes-the-ai-analyst`) so I cannot push to its branch from the keboola org. The triple-surface additions are non-trivial (2 new files + 4 modified), and the Devin BUG_0001 fix changes the same coverage-gate test, so a follow-up PR would create a noisy merge ordering. Easier to land both in one cohort here and close #565 with a pointer to this one.

## Project policy — why three surfaces

Every public REST endpoint must be reachable from three surfaces:

  1. **REST (HTTP)** — `app.web.router`
  2. **CLI (`agnes …`)** — `cli/commands/…`
  3. **MCP tool** — `app.api.mcp_http`

The three audiences (operator dashboards, analyst CLI scripts, agent / Claude Desktop tooling) each strand on their own surface if coverage is one-sided. Existing REST endpoints are NOT retroactively gated — the policy is a forward-only ratchet, and `tests/test_documentation_api_triple_surface.py` locks the v0.68.0 cohort (just `/documentation/api` for now). When future endpoints land that ought to live on all three, the cohort grows and the ratchet enforces parity.

## Adds on top of #565

  - **`cli/commands/docs.py`** — `agnes docs api` renders the same `docs/api-reference.md` in the terminal via Rich's Markdown formatter. Wired into `cli/main.py`.
  - **`app/api/mcp_http.py`** — new `documentation_api` MCP tool returns the raw Markdown so Claude Desktop / Cursor / Cline can look up the REST surface without leaving the chat.
  - **`tests/test_documentation_api_triple_surface.py`** — forward-only ratchet that locks the cohort by walking the REST router, the CLI command tree, and the MCP tool registry.

## Devin BUG_0001 fix

#565's `path not in doc` substring check at `tests/test_api_docs_coverage.py:46` false-passes when a new endpoint is a prefix of an already-documented one (e.g. removing `/api/health` would still satisfy the gate via `/api/health/detailed`). Replaced with a token-bounded regex (`re.escape(path) + r"(?![A-Za-z0-9/_\-{}])"`) and locked by `test_path_documented_helper_rejects_prefix_overlap`.

## Misc

- `docs/api-reference.md` top-of-page note describes the three surfaces.
- CHANGELOG `[0.68.0]` entry rewritten to cover all three; the orphaned Cowork bullet from `[Unreleased]` (PR #488 merged on main without a release-cut) is rolled into 0.68.0 too so it doesn't stay orphaned.
- Version stays `0.68.0` — Petr's pre-cut version. No bump needed.

## Tests

```
tests/test_api_docs_coverage.py::test_doc_exists PASSED
tests/test_api_docs_coverage.py::test_every_public_api_path_documented PASSED
tests/test_api_docs_coverage.py::test_path_documented_helper_rejects_prefix_overlap PASSED
tests/test_documentation_api_triple_surface.py::test_rest_endpoints_callable PASSED
tests/test_documentation_api_triple_surface.py::test_cli_subcommands_registered PASSED
tests/test_documentation_api_triple_surface.py::test_mcp_tools_registered PASSED
```

---

<a href="https://app.devin.ai/review/keboola/agnes-the-ai-analyst/pull/566" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: app/web/templates/_app_header.html [hunk 1]
```diff
@@ -47,7 +47,7 @@
...
+        {% set _admin_active = _path.startswith('/admin/tables') or _path.startswith('/admin/tokens') or _path.startswith('/admin/users') or _path.startswith('/admin/groups') or _path.startswith('/admin/access') or _path.startswith('/admin/server-config') or _path.startswith('/admin/agent-prompt') or _path.startswith('/admin/workspace-prompt') or _path.startswith('/admin/marketplaces') or _path.startswith('/admin/mcp-sources') or _path.startswith('/admin/mcp-tools') or _path.startswith('/admin/store') or _path.startswith('/admin/scheduler-runs') or _path.startswith('/admin/activity') or _path.startswith('/admin/telemetry') or _path.startswith('/admin/usage') or _path.startswith('/admin/sessions') or _path.startswith('/admin/corporate-memory') or _path.startswith('/admin/sync') or _path.startswith('/documentation') %}
```
[the remark this answers is not in the document]
Comment (ZdenekSrotyr) [reply app/web/templates/_app_header.html:50]: Pre-existing as Devin notes, out of scope for #566's triple-surface coverage focus. Tracked separately as nav-active follow-up. Resolving.

File: tests/test_api_docs_coverage.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    The substring check (``path in doc``) silently false-passes when an
+    endpoint is a prefix of an already-documented one — e.g. removing
+    ``/api/health`` from the doc while ``/api/health/detailed`` remains
+    would still report "documented" because ``/api/health`` is a substring
+    of the surviving path. Anchor on a trailing non-path character so the
+    next character must end the token (whitespace, backtick, pipe, comma,
+    paren, end-of-line) — never another path segment.
+    """
+    pattern = re.escape(path) + r"(?![A-Za-z0-9/_\-{}])"
```
[the remark this answers is not in the document]
Comment (ZdenekSrotyr) [reply tests/test_api_docs_coverage.py:49]: Acknowledged — minor edge case (dotted paths aren't an Agnes API pattern in practice; all current /api/* routes are slash-separated). Will fold into the next coverage-gate touch if dotted siblings ever emerge. Resolving.
